libraries/Microsoft.Bot.Builder/BotState.cs

```diff
@@ -142,6 +143,23 @@ public virtual async Task DeleteAsync(ITurnContext turnContext, CancellationToke
             await _storage.DeleteAsync(new[] { storageKey }, cancellationToken).ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// Returns the raw cached data from the TurnContext.
```

Comment:
I'm not clear, reading this, aboput the behavior. 

If I poke at the JToken that comes back, am I poking at a clone of the data, or at the underlying data itself? I think documenting the semantics here are worthwhile.

Reply:
... and equally important, the semantics of the returned JSON should match the JS SDK Behavior.

Reply:
What do you mean by "the semantics returned JSON should match the JS SDK Behavior" the shape of the JSON is different - that's the nature of the beast?
Serialization is an antipattern, but it is the pattern adopted in this SDK (against my advice).
Anyhow, academic, this makes writing tings like inspection easier. Its a view on internal state.
Technically the JavaScript is not a copy and this is. There is no such thing as a non-leaky abstraction.

Reply:
I'll add a comment.
this is now at parity. and the constant "state" is hidden - as it is in the JavaScript